crates/miden-protocol/src/note/inputs.rs

```diff
@@ -145,6 +145,15 @@ impl Serializable for NoteInputs {
         target.write_u16(values.len().try_into().expect("inputs len is not a u16 value"));
         target.write_many(values);
     }
+
+    fn get_size_hint(&self) -> usize {
+        // Size of the serialized inputs length prefix.
+        let u16_size = 0u16.get_size_hint();
+
+        let values_size: usize = self.values.iter().map(|value| value.get_size_hint()).sum();
```

Comment:
Why do we need to iterate over inputs to compute the size? Each input is just one `Felt` and so we should be able to just multiple the number of inputs by felt size, right?